### PR TITLE
Update: Length for materialQuantities in RevitConverter

### PR DESCRIPTION
**Pull request by ks-cph**
## Description & motivation

- Add length of elements to MaterialQuantities
- Allows for query of material quantities
- fixes issue #1944 

## Changes:

- ```Objects.Converter.Revit.ConverterRevit```
- ```Objects.Other.MaterialQuantity```

## Screenshots:

Two differnet elements before(l.) and after(r.)

## Validation of changes:
I have testes it with Revit projects and different element categories.

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

**Review by teocomi — CHANGES_REQUESTED** [review]
Thanks @ks-cph , this looks good.
Although since we're working with release branches, this would need to be started from the `release\2.11` branch and merged there, instead of using `main`.
Can you please do so?

File: Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMaterialQuantities.cs [hunk 1]
```diff
@@ -25,16 +25,18 @@ public Objects.Other.MaterialQuantity MaterialQuantityToSpeckle(DB.Element eleme
       //Get quantities
       double volume = element.GetMaterialVolume(material.Id);
       double area = element.GetMaterialArea(material.Id, false); //To-Do: Do we need Paint-Materials
-
+                                                                 //length
+      double length = 0;
+      if (LocationToSpeckle(element) is ICurve) length = (LocationToSpeckle(element) as ICurve).length/1000;  //LocationCurve default is mm (Speckle)
```
Comment (AlanRynne) [inline Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMaterialQuantities.cs:30]: Not sure about this line. I'll assign @connorivy to give this a quick check before going in.

**Review by AlanRynne — COMMENTED** [review]
Added my thoughts on a suspicious line of code. 🙂

**Comment by connorivy** [comment]
Hey @ks-cph,

Thank you very much for the PR! Also, I apologize that it took so long to get around to. Everything looks really good except for that line that Alan previously pointed out and the following few lines. It wasn't a huge issue by any means, but it seems that the units were hard coded to be meters. 

Perhaps you were just trying to save me from using the imperial system of units, and for that I thank you, but sometimes I am intent on making my life more difficult by using those units. So I made another PR (https://github.com/specklesystems/speckle-sharp/pull/2025) that I started from your code where I just send using the default units of the speckle commit whatever those may be. 

Also, I noticed that the length prop was not used by the majority of the elements that I happened to be sending at the time when I was testing this, so I ended up just adding the length prop dynamically only when applicable instead of adding it to the materialQuantities object.

Could you take a look at this pr and see if it still works for what you are trying to do? If not then please let me know what doesn't work for you and we can talk about a solution.

**Pull request closed without merging**